sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -726,6 +726,14 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   }
 
   /**
+   * Return the value of an optional Spark SQL configuration property for the given key. If the key
+   * is not set yet, returns None.
+   */
+  def getOptionalConf[T](entry: OptionalConfigEntry[T]): Option[T] = {
```

Comment:
Ideally its an `OptionalConf` so getConf should return `Option[T]`, but this will change the current behaviour(we won't be able to throw proper errors if the key is missing).
In that case what we can do instead is something like this
```scala
getConf(CHECKPOINT_LOCATION).getOrElse(throw new NoSuchElementException(CHECKPOINT_LOCATION.key))
```
Suggestions ?